envm/Dockerfile

```diff
@@ -1,46 +1,325 @@
+    echo "Deleting conda install script" 1>&2 && \
+      rm mambaforge.sh && \
+    echo "Update mamba" 1>&2 && \
+      /miniforge/bin/mamba update --quiet --yes \
+        --name base \
+        --channel conda-forge \
+        mamba && \
+    echo "Install environment with mamba" 1>&2 && \
+      /miniforge/bin/mamba env create --quiet \
+        --file /tmp/environment.yaml && \
+    echo "Cleaning up conda" && \
+      /miniforge/bin/conda clean --all --quiet --yes
 
-#get WTS report's RMD and R params script
+# Add in RMD files
+# Add in RMD files
 COPY rmd_files /rmd_files
 
-#set working directory
-WORKDIR /rmd_files
+# Add in the Rprofile.site file
+RUN mv "/rmd_files/Rprofile.site" \
+       "${R_INSTALL_PATH}/lib/R/etc/Rprofile.site"
 
-#change permissions
-#RUN chmod +x /rmd_files/*
-RUN chmod -R 777 /rmd_files
+# Install BiocManager and renv
+RUN echo "Install Renv and BiocManager packages though R" 1>&2 && \
+      Rscript -e 'options(Ncpus=2L, repos="https://cloud.r-project.org/"); \
+                  install.packages(c( "BiocManager", "renv", "yaml"), \
+                                   quiet=TRUE)'
 
-# Setting locales and timezones, based on https://github.com/jacksoncage/node-docker/blob/master/Dockerfile
-# (setting UTC for readr expecting UTC https://rdrr.io/github/tidyverse/readr/src/R/locale.R - credit umccrise)
-ENV LANGUAGE en_US.UTF-8
-ENV LANG en_US.UTF-8
-ENV LC_ALL en_US.UTF-8
-RUN apt-get update && \
-    apt-get install -y locales language-pack-en && \
-    locale-gen en_US.UTF-8 && \
-    dpkg-reconfigure locales && \
-    apt-get install -y tzdata && \
-    echo "Etc/UTC" > /etc/timezone && \
-    dpkg-reconfigure -f noninteractive tzdata
+# Biocmanager pacakges that need to be installed prior to renv being activated
+# See https://github.com/rstudio/renv/issues/934 for more details
+# We've also added in rmd_files/renv_required_libraries.R
+# To make sure that libraries that aren't auto downloaded by renv are added to the list
+RUN echo "Install Bioconductor packages thorugh R" 1>&2 && \
+      Rscript -e 'options(repos=BiocManager::repositories()); \
+                  BiocManager::install(ask=FALSE, pkgs=c( \
+                                                         "EDASeq", \
+                                                         "limma", \
+                                                         "preprocessCore", \
+                                                         "edgeR", \
+                                                         "rhdf5", \
+                                                         "tximport", \
+                                                         "EnsDb.Hsapiens.v75", \
+                                                         "EnsDb.Hsapiens.v86", \
+                                                         "BSgenome.Hsapiens.UCSC.hg19", \
+                                                         "BSgenome.Hsapiens.UCSC.hg38" ), \
```

Comment:
are these the only renv dependencies? Couldn't find this info in the issue linked.

Reply:
The dependencies that aren't explicitly specified are in [this file](https://github.com/umccr/RNAsum/blob/dockerfile-updates/rmd_files/renv_required_libraries.R). The issue linked likely doesn't correlate to the current issue. The reasons these packages needed to be installed locally was due to the way renv was trying to install them. I can try and emulate this at some point so its resolved and we don't have to specify many of the Bioc packages first.

Reply:
Okay, to summarise - there are couple dependencies that are installed via mamba (plotly-orca, libopenblas==0.3.7) and renv dependencies are installed separately as listed in this [file](https://github.com/umccr/RNAsum/blob/dockerfile-updates/rmd_files/renv_required_libraries.R). 
Why are these dependencies (line 233-242) not part of the linked file?

Reply:
> The installation of bioconductor packages inside the dockerfile are for packages that renv has trouble installing.

I'd add this as a comment to docker file. So that next time we hit a package that renv is having trouble installing we can add to this bioconductor installation command. 

> They need to be inside the rmd_files directory
What is `they` here?